src/net_processing.cpp

```diff
@@ -847,6 +847,30 @@ void PeerLogicValidation::UpdatedBlockTip(const CBlockIndex *pindexNew, const CB
     nTimeBestReceived = GetTime();
 }
 
+namespace {
+
+void HandleBlockDoS(std::string node_name, CNode * const pfrom, const int nDoS, const bool is_header) {
+    // We never actually DoS ban for invalid blocks, merely disconnect nodes if we're relying on them as a primary node
+    NodeId node_id = pfrom->GetId();
+    if (node_name.empty()) {
```

Comment:
IIRC this is the IP of the peer. The point of accepting it as a parameter is to avoid the lock when we already have it.